website/docs/r/vapp_vm.html.markdown

```diff
@@ -175,3 +178,12 @@ its parameters do force recreation of VMs!
   * `ip_allocation_mode=MANUAL` - **`ip`** value must be valid IP address from a subnet defined in `static pool` for network.
 
   * `ip_allocation_mode=NONE` - **`ip`** field can be omitted or set to an empty string "". Empty string may be useful when doing HCL variable interpolation.
+
+<a id="customization"></a>
+## Customization
+
+* `force` (Optional) **Warning.** `true` value will cause the VM to reboot on every `apply` operation.
+This field works as a flag and triggers force customization when `true` during an update 
+(`terraform apply`) every time. It never complains about a change in statefile. Can be used when guest customization
+is needed after VM configuration (e.g. NIC change, customization options change, etc.) and then set back to `false`.
+**Note.** It will not have effect when `power_on` field is set to `false`.
```

Comment:
Maybe we should throw warning from the code? @lvirbalas

Reply:
We discussed this with Dainius Yesterday. Yes, we need, and ideally during *plan*, if possible, because it will rebooting VM.

Reply:
I have added a Terraform validator which emits warning.
Here's how the `terraform plan` looks like (only when `customization.force=true`):
```bash
...
      + template_name                  = "LB"
      + vapp_name                      = "lb-test-web"
      + vdc                            = "tf_vdc"

      + customization {
          + force = true
        }
    }

Plan: 2 to add, 0 to change, 0 to destroy.

Warning: Using 'true' value for field 'vcd_vapp_vm.customization.force' will reboot VM on every 'terraform apply' operation



  on main.tf line 62, in resource "vcd_vapp_vm" "web":
  62: resource "vcd_vapp_vm" "web" {



------------------------------------------------------------------------

Note: You didn't specify an "-out" parameter to save this plan, so Terraform
can't guarantee that exactly these actions will be performed if
"terraform apply" is subsequently run.
```

When doing `terraform apply` however these warnings are not shown. I think I'm ok with that, but need other opinions.

The thing I like about using built in validator for warning is that it prints very nicely and points out location automatically while if I add a print to stdout manually I can't control it so nicely.

Reply:
Adding output for `terraform validate` as well:
```bash
$  terraform validate

Warning: Using 'true' value for field 'vcd_vapp_vm.customization.force' will reboot VM on every 'terraform apply' operation



  on main.tf line 62, in resource "vcd_vapp_vm" "web":
  62: resource "vcd_vapp_vm" "web" {


Success! The configuration is valid, but there were some validation warnings as shown above.

```

Reply:
Perfect. Does this get printed with `apply` without --auto-approve option too?

Reply:
No. That is the problem as I mentioned above.

> When doing `terraform apply` however these warnings are not shown. I think I'm ok with that, but need other opinions.

Reply:
You can write warning on apply phase using functionality from upload resource if we want that.

Reply:
Yes. I could, but the somewhat less nice thing is that it would be "somewhere in the middle" while the warning in validator prints it very clearly.